pkg/component/etcd/etcd/crd.go

```diff
@@ -9,98 +9,146 @@ import (
 		return err
 	}
 
 	if len(etcdList.Items) > 0 {
-		return fmt.Errorf("cannot delete etcd CRDs because there are still druidv1alpha1.Etcd resources left in the cluster")
+		return fmt.Errorf("cannot delete etcd CRDs because there are still druidcorev1alpha1.Etcd resources left in the cluster")
 	}
 
-	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: etcdCRDName}}); client.IgnoreNotFound(err) != nil {
+	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: druidcorecrds.ResourceNameEtcd}}); client.IgnoreNotFound(err) != nil {
 		return err
 	}
 
-	etcdCopyBackupsTaskList := &druidv1alpha1.EtcdCopyBackupsTaskList{}
+	etcdCopyBackupsTaskList := &druidcorev1alpha1.EtcdCopyBackupsTaskList{}
 	if err := c.client.List(ctx, etcdCopyBackupsTaskList); err != nil && !meta.IsNoMatchError(err) && !apierrors.IsNotFound(err) {
 		return err
 	}
 
 	if len(etcdCopyBackupsTaskList.Items) > 0 {
-		return fmt.Errorf("cannot delete etcd CRDs because there are still druidv1alpha1.EtcdCopyBackupsTask resources left in the cluster")
+		return fmt.Errorf("cannot delete etcd CRDs because there are still druidcorev1alpha1.EtcdCopyBackupsTask resources left in the cluster")
 	}
 
-	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: etcdCopyBackupsTaskCRDName}}); client.IgnoreNotFound(err) != nil {
+	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: druidcorecrds.ResourceNameEtcdCopyBackupsTask}}); client.IgnoreNotFound(err) != nil {
 		return err
 	}
 
 	var fns []flow.TaskFn
 
-	for _, resource := range crdResources {
+	for _, resource := range c.crdGetter.GetAllCRDs() {
 		r := resource
 		fns = append(fns, func(ctx context.Context) error {
-			return client.IgnoreNotFound(c.applier.DeleteManifest(ctx, kubernetes.NewManifestReader([]byte(r))))
+			return client.IgnoreNotFound(c.client.Delete(ctx, r, &client.DeleteOptions{}))
 		})
 	}
 
 	return flow.Parallel(fns...)(ctx)
 }
+
+// CRDGetter provides methods to get CRDs defined in etcd-druid.
+type CRDGetter interface {
+	// GetAllCRDs returns a map of CRD names to CRD objects
+	GetAllCRDs() map[string]*apiextensionsv1.CustomResourceDefinition
+	// GetCRD returns the CRD with the given name.
+	// An error is returned if no CRD is found with the given name.
+	GetCRD(name string) (*apiextensionsv1.CustomResourceDefinition, error)
+}
+
+type _crdGetter struct {
+	crdResources map[string]*apiextensionsv1.CustomResourceDefinition
+}
+
+var _ CRDGetter = (*_crdGetter)(nil)
+
+// NewCRDGetter creates a new CRDGetter.
+func NewCRDGetter(k8sVersion *semver.Version) (CRDGetter, error) {
+	crdResources, err := getEtcdCRDS(k8sVersion)
```

Comment:
```suggestion
	crdResources, err := getEtcdCRDs(k8sVersion)
```

For consistency with https://github.com/gardener/gardener/blob/a7037096c8510162f48c159d021d06c1e450e6d6/pkg/component/observability/logging/fluentoperator/crds.go#L47, https://github.com/gardener/gardener/blob/a7037096c8510162f48c159d021d06c1e450e6d6/pkg/component/observability/monitoring/prometheusoperator/crd.go#L41 and `GetAllCRDs` func below.